Source/SwiftLintCore/Models/Location.swift

```diff
@@ -23,8 +23,17 @@ public struct Location: CustomStringConvertible, Comparable, Codable, Sendable {
 
     /// The file path for this location relative to the current working directory.
     public var relativeFile: String? {
-        let url = URL(fileURLWithPath: FileManager.default.currentDirectoryPath, isDirectory: true)
-        return file?.replacingOccurrences(of: url.filepath + "/", with: "")
+        guard let file else { return nil }
+        let filePath = URL(fileURLWithPath: file).filepath
+        var basePath = URL(fileURLWithPath: FileManager.default.currentDirectoryPath, isDirectory: true).filepath
+        if !basePath.hasSuffix("/") {
+            basePath += "/"
+        }
+
+        if filePath.hasPrefix(basePath) {
+            return String(filePath.dropFirst(basePath.count))
+        }
+        return file
```

Comment:
Sorry, I should've been more clear. We should de-compose the path with `pathComponents` based handling. we would need to ensure that the CWD is entirely present though.

Reply:
Almost - but rather than the `.joined(separator: "/")`, use a reduction of `appendingPathComponent` (or if we can use newer APIs, `appending(components:)`).